ruby.c

```diff
@@ -1798,6 +1798,26 @@ copy_str(VALUE str, rb_encoding *enc, bool intern)
     return rb_enc_interned_str(RSTRING_PTR(str), RSTRING_LEN(str), enc);
 }
 
+#if USE_YJIT
+// Check that an environment variable is set to a truthy value
+static bool
+env_var_truthy(const char *name)
+{
+    const char *value = getenv(name);
+
+    if (!value)
+        return false;
+    if (strcmp(value, "1") == 0)
+        return true;
+    if (strcmp(value, "true") == 0)
+        return true;
+    if (strcmp(value, "yes") == 0)
+        return true;
+
+    return false;
```

Comment:
Could we also check for caps values?

```suggestion
    return (strcmp(value, "1") == 0))
      && (strcmp(value, "true") == 0))
      && (strcmp(value, "TRUE") == 0))
      && (strcmp(value, "yes") == 0))
      && (strcmp(value, "YES") == 0));
```

Reply:
Your suggestion looks good to me. Why don't you use strcasecmp() instead?

However, please note that this pull request is intended for backporting. As per our policy for maintaining CRuby's stable branch, we primarily backport changesets from the master branch. Could you create a pull request for the master branch instead?
Thanks.